### PR TITLE
Technical setup of an Angular application

### PR DESCRIPTION
**Pull request by wendlans**
# Summary
I've created a simple Angular application (`app.component` is still a placeholder and contains the pre-generated content).
Also, I've added some basics:
* Error handling including three base error types: `Silent`, `Recoverable` and `Fatal`
* Base state module + `immer`
* A constants file containing some route parameters
* A page enum containing the name of the main pages.
* Eslint 9 including some rules
* Pre-Hooks to check branch-name and commit-msg; also to run `prettier` automatically

Please note that the technical setup is still ongoing. However, I've still want to merge this back to main to have a common base to continue developing together.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+trim_trailing_whitespace = true
+
+[*.ts]
+quote_type = single
+ij_typescript_use_double_quotes = false
+max_line_length = 140
+
+[*.md]
+max_line_length = off
+trim_trailing_whitespace = false
```
Comment (SanseroGames) [inline .editorconfig:18]: I'm curious, what is the reason to not trim trailing whitespace in markdown files?

File: .lintstagedrc [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "*": "prettier --write -u"
```
Comment (SanseroGames) [inline .lintstagedrc:2]: Shouldn't eslint be added here as well?

File: src/app/error-handling/components/fatal-error/fatal-error.component.scss [hunk 3]
```diff
@@ -0,0 +1,8 @@
+.fatal-error-page {
+  background-color: red;
+  color: white;
+
+  fatal-error-page__error-message {
```
Comment (SanseroGames) [inline src/app/error-handling/components/fatal-error/fatal-error.component.scss:5]: This should be a class.

File: .editorconfig (hunk 1 above)
Comment (wendlans) [reply .editorconfig:18]: Well spotted!

I've quickly checked this in another project. And we can definitely use `trim_trailing_whitespace = true` here. I'll change it.

File: .lintstagedrc (hunk 2 above)
Comment (wendlans) [reply .lintstagedrc:2]: I'll add it using `eslint --fix`. We'll see if that causes any problems.

File: src/app/error-handling/components/fatal-error/fatal-error.component.scss (hunk 3 above)
Comment (wendlans) [reply src/app/error-handling/components/fatal-error/fatal-error.component.scss:5]: Eagle eye strikes again. Thanks! I'll fix it.

**Pull request merged**